test/helpers/position_assertions.h

```diff
@@ -49,9 +50,15 @@ class RangeAssertion {
     static std::vector<std::shared_ptr<ErrorAssertion>>
     getErrorAssertions(const std::vector<std::shared_ptr<RangeAssertion>> &assertions);
 
+    static std::vector<std::shared_ptr<ErrorAssertion>>
+    allTypedAsErrorAssertions(const std::vector<std::shared_ptr<RangeAssertion>> &assertions);
```

Comment:
Is this for all assertions that show up directly as `error:` in a test file? Could you add a comment to explain how this differs from `getErrorAssertions`?